src/rust/bitbox02-rust/src/platform/bitboxbase/config.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019 Shift Cryptosecurity AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::fmt::Write;
+use core::time::Duration;
+
+use arrayvec::ArrayString;
+
+use crate::c_char;
+use crate::error::Error;
+use crate::util::{FixedCString, Ipv4Addr};
+
+/// This determines if the LED should be turned off in some states.
+#[derive(Debug, Clone)]
+#[repr(u8)]
+#[allow(dead_code)] // We are setting this with transmute
+pub enum StatusLedMode {
+    /// Always show the LED
+    Always,
+    /// Only show the led in the Working, Warning and Error states
+    OnWorking,
+    /// Only show the led in the Warning and Error states
+    OnWarning,
+    /// Only show the led in the Error state
+    OnError,
+}
+
+#[derive(Debug, Clone)]
+#[repr(u8)]
+#[allow(dead_code)] // We are setting this with transmute
+pub enum StatusScreenMode {
+    /// Always show the screen
+    Always,
+    /// Only show the screen in the Working, Warning and Error states
+    OnWorking,
+    /// Only show the screen in the Warning and Error states
+    OnWarning,
+    /// Only show the screen in the Error state
+    OnError,
+}
+
+pub struct Config {
+    pub(crate) status_led_mode: StatusLedMode,
+    pub(crate) status_screen_mode: StatusScreenMode,
+    pub(crate) default_display_duration: Duration,
+    pub(crate) hostname: Option<ArrayString<[u8; 64]>>,
+    pub(crate) ip: Option<Ipv4Addr>,
+}
+
+impl Config {
+    pub const fn new() -> Config {
+        Config {
+            status_led_mode: StatusLedMode::Always,
+            status_screen_mode: StatusScreenMode::OnWarning,
+            default_display_duration: Duration::from_secs(10),
+            hostname: None,
+            ip: None,
+        }
+    }
+
+    pub fn set_status_led_mode(&mut self, mode: StatusLedMode) {
+        self.status_led_mode = mode;
+    }
+
+    pub fn set_status_screen_mode(&mut self, mode: StatusScreenMode) {
+        self.status_screen_mode = mode;
+    }
+
+    pub fn set_hostname(&mut self, hostname: &str) -> Result<(), Error> {
+        if !hostname.is_ascii() {
+            return Err(Error::InvalidHostname);
+        }
+        if hostname.len() < 1 || hostname.len() > 63 {
+            return Err(Error::InvalidHostname);
+        }
+        // Since we checked that the string contains at least one character we can safely unwrap
+        if hostname.chars().nth(0).unwrap() == '-' || hostname.chars().last().unwrap() == '-' {
```

Comment:
have you checked if this is equivalent to the regexp used by the base middleware and app? should have the same check in both.

Reply:
Nope didn't check. I follow the RFC. I hope you do to in the MW. https://tools.ietf.org/html/rfc1123#page-13

Reply:
@Tomasvrba please check

Reply:
The middleware and app regex check is more restrictive: `^[a-z][a-z0-9-]{0,22}[a-z0-9]$`,  does not allow hostnames such as: `a.a`, `a.1`, or `1a` which should be valid according to the RFC and would need a check more like this:   
`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`

@Stadicus, we had a check previously that did allow numbers as the first character, but it was later changed. Do you recall why? 
According to the RFC:
> One aspect of host name syntax is hereby changed: the
      restriction on the first character is relaxed to allow either a
      letter or a digit.  Host software MUST support this more liberal
      syntax.

Reply:
No, I don't remember. We can allow it if it's ok according to RFC. Just need to update everywhere.

Reply:
A valid hostname is `<let-or-digit>[*[<let-or-digit-or-hyphen>]<let-or-digit>]` (1 to 63 chars)

Reply:
See page 5 https://tools.ietf.org/html/rfc952 and page 13 https://tools.ietf.org/html/rfc1123#page-13

Reply:
The HSM as it is written right now only accepts the first **label** of a hostname. That means **home** in **home.example.com**.

Reply:
https://en.wikipedia.org/wiki/Hostname